### PR TITLE
add UTs for submit

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
- add UTs for submit, both standard and distillation input scripts.
- [fix the case when numb_models does not match number of init models](https://github.com/deepmodeling/dpgen2/commit/ca4372b8a1c05099e80ee252e1c06b4037e7c04a)

### CODE REVIEW

File: dpgen2/entrypoint/args.py
Comment (Angel-Jia) [inline dpgen2/entrypoint/args.py:81]: It seems that `init_models_paths` cannot be set to `null` or `[]` if no init_model is available. I‘m not sure if it is in line with expectations.

File: dpgen2/entrypoint/args.py
Comment (njzjz) [inline dpgen2/entrypoint/args.py:79]:
```suggestion
            [list, None],
```

File: dpgen2/entrypoint/submit.py
Comment (wanghan-iapcm) [inline dpgen2/entrypoint/submit.py:380]: @Angel-Jia Please see the check here. It does not make sense if the length of the list does not matches number of models. therefore, we raise error when an empty list is provided.

File: dpgen2/entrypoint/args.py
Comment (wanghan-iapcm) [reply dpgen2/entrypoint/args.py:79]:
```python
from dargs import Argument

def make_arguments():
    arg_foo = Argument("foo", int, optional=True, default=None)
    return [arg_foo]

base = Argument("base", dict, make_arguments())
data = base.normalize_value({'foo' : None})
print(data)
data = base.normalize_value({})
print(data)
```
the result is 
```
{'foo': None}
{'foo': None}
```

Comment (njzjz) [reply dpgen2/entrypoint/args.py:79]: https://github.com/deepmodeling/dargs/blob/11d3fc9ec197924534d9612a155dbfa9d3368583/dargs/dargs.py#L214

This is because the type of default will be added to dtype automatically. However, it's better to explicitly set it.

**Review by Angel-Jia — APPROVED** [review]

File: dpgen2/entrypoint/args.py
Comment (wanghan-iapcm) [reply dpgen2/entrypoint/args.py:79]: In this project we usually write 
```
Optional[int] = None
```
as type hint and default value, not 
```
Union[int, None] = None
```
The dargs definition is more consistent with the first way.

**Comment by wanghan-iapcm** [comment]
@Angel-Jia please do not merge the PR until all reviewers approve.  (This PR has not been approved by @njzjz )

**Pull request merged**
